src/main.rs

```diff
@@ -63,27 +63,57 @@ fn main() {
             "trace" => LevelFilter::Trace,
             "off" => LevelFilter::Off,
             level => {
-                eprintln!(
+                eprintln!( // Cannot use logging error-macro before logging is initialised
                     "Log-level was not in {{error, warn, info, debug, trace, off}}, received: {:?}",
                     level
                 );
                 exit(1);
             }
         },
     )) {
-        model_check(args); // not sure whether this requires match on Result
+        match args.subcommand() {
+            ("solver", Some(solver_args)) => {
+                match solver_args.value_of("model_type").unwrap() {
+                    "json" => {
+                        model_check(solver_args);
+                        ()
+                    }
+                    "lcgs" => { () }
+                    _ => { () }
+                }
+            }
+            ("graph", Some(graph_args)) => (),
+            _ => (),
+        }
     }
 }
 
-fn model_check(args: ArgMatches) -> Result<(), Box<dyn Error>> {
+fn model_check(args: &ArgMatches) -> Result<(), Box<dyn Error>> {
     info!("Model checking on formula: {:?}", args.value_of("formula"));
 
-    let mut file = File::open(args.value_of("json_model").unwrap())?;
-    let mut game_structure = String::new();
-    file.read_to_string(&mut game_structure)?;
-    let game_structure: EagerGameStructure = serde_json::from_str(game_structure.as_str())?;
-
-    let mut file = File::open(args.value_of("json_formula").unwrap())?;
+    let game_structure = match args.value_of("model_type") {
+        Some("lcgs") => {
+            // TODO: implement lgcs parser
+            EagerGameStructure {
+                player_count: 0,
+                labeling: vec![],
+                transitions: vec![],
+                moves: vec![],
+            }
+        }
```

Comment:
You have already checked `model_type` in main. Why do it here again?

Reply:
Because I was unsure where we'll make the lcgs parsing. Fixed in af880a9